### PR TITLE
ecf-1-524 Lead providers can choose delivery partners for a school

### PR DESCRIPTION
**Pull request by amazimbe**
### Context
DfE need Lead Providers to specify which delivery partner is assigned to which school so that schools don't mistakenly challenge the partnership because they only know about the delivery partner.

### Changes proposed in this pull request
A form with radio buttons that select the delivery partner for a school

### Review Checks
- [x] All pages have automated accessibility checks via cypress
- [x] All pages have visual tests via cypress + percy
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
Login as lead-provider@example.com. From the dashboard click on find schools
image (2)

### CODE REVIEW

File: app/controllers/lead_providers/report_schools_controller.rb [hunk 1]
```diff
@@ -3,5 +3,24 @@
...
+      @delivery_partners = current_user.lead_provider.delivery_partners
+    end
+
+    def check_delivery_partner
+      @delivery_partner_form = LeadProviderDeliveryPartnerForm.new(
+        params.require(:lead_provider_delivery_partner_form).permit(:delivery_partner_id),
+      )
+
+      if @delivery_partner_form.valid?
+        session[:delivery_partner_id] = @delivery_partner_form.delivery_partner_id
```
Comment (kicferk1) [inline app/controllers/lead_providers/report_schools_controller.rb:18]: I would be tempted to make the form serializable and do `@delivery_partner_form.serializable_hash` here.

Comment (kicferk1) [reply app/controllers/lead_providers/report_schools_controller.rb:18]: Also, should we wipe the session in `start` action?

File: app/views/lead_providers/report_schools/choose_delivery_partner.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <div class="govuk-grid-column-two-thirds">
+    <h1 class="govuk-heading-xl">Choose the delivery partner</h1>
+    <p>Select the delivery partner that these schools have confirmed an agreement with.</p>
+    <%= form_with model: @delivery_partner_form, url: check_delivery_partner_lead_providers_report_schools_path, method: :post do |f| %>
+      <%= f.govuk_collection_radio_buttons(
+        :delivery_partner_id,
+        @delivery_partners,
+        :id,
+        :name,
+        legend: nil) %>
```
Comment (kicferk1) [inline app/views/lead_providers/report_schools/choose_delivery_partner.html.erb:13]: Hmm, I wonder if we should make the paragraph above into the `legend`.

File: spec/cypress/integration/lead_providers/ReportSchools.feature [hunk 3]
```diff
@@ -10,3 +10,14 @@ Feature: Report Schools flow
     And "page body" should contain "2021"
     And the page should be accessible
     And percy should be sent snapshot called "Lead provider report schools start page"
+
+  Scenario: Selecting a delivery partner
+    Given scenario "lead_provider_with_delivery_partners" has been run
```
Comment (kicferk1) [inline spec/cypress/integration/lead_providers/ReportSchools.feature:15]: I think that's something I would put in `Background` part - hopefully all tests in this spec will need a setup like that.

**Review by kicferk1 — COMMENTED** [review]
A few small comments

File: app/controllers/lead_providers/report_schools_controller.rb (hunk 1 above)
Comment (amazimbe) [reply app/controllers/lead_providers/report_schools_controller.rb:18]: Definitely agree with the comment about the session. I can't see any advantage to making the form serializable though. If there were many fields and / or I was using this to initialise an object then may be.

File: app/views/lead_providers/report_schools/choose_delivery_partner.html.erb (hunk 2 above)
Comment (amazimbe) [reply app/views/lead_providers/report_schools/choose_delivery_partner.html.erb:13]: I could but these legends use a header tag by default and I find what we've been doung to use a p tag quite fiddly and difficult to understand when reading the code without offering any clear benefit.

File: app/controllers/lead_providers/report_schools_controller.rb (hunk 1 above)
Comment (ethanmills) [reply app/controllers/lead_providers/report_schools_controller.rb:18]: My expectation is that we will use a single form model for this step and the next step of selecting schools, either via CSV or UI. The changes that requires would be part of the later work though

File: spec/cypress/app_commands/scenarios/lead_provider_with_delivery_partners.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+profile = LeadProviderProfile.last
```
Comment (ethanmills) [inline spec/cypress/app_commands/scenarios/lead_provider_with_delivery_partners.rb:3]: We need some ordering here

File: app/controllers/lead_providers/report_schools_controller.rb (hunk 1 above)
Comment (amazimbe) [reply app/controllers/lead_providers/report_schools_controller.rb:18]: @ethanmills I don't see how we can use a single form model for this part and the csv upload part. Those two things are just not related logically.

**Comment by amazimbe** [comment]
[Select a delivery partner](https://trello.com/c/3fW2dDz7/524-select-a-delivery-partner)

**Review by kicferk1 — APPROVED** [review]

**Comment by kicferk1** [comment]
Some conflicts with Ethan's error pages that just went in I think

**Pull request merged**